orca-igor/src/main/groovy/com/netflix/spinnaker/orca/igor/pipeline/CIStage.java

```diff
@@ -53,11 +53,15 @@ public void taskGraph(@Nonnull Stage stage, @Nonnull TaskNode.Builder builder) {
   }
 
   private boolean waitForCompletion(Stage stage) {
-    String waitForCompletion = (String) stage.getContext().get("waitForCompletion");
-    if (waitForCompletion != null && waitForCompletion.equals("false")) {
-      return false;
+    Object contextValue = stage.getContext().get("waitForCompletion");
```

Comment:
As an aside, you could avoid this with a `stage.mapTo()` and letting jackson handle proper type deserialization.

Reply:
Switched to `mapTo()` and added tests around this in #2768